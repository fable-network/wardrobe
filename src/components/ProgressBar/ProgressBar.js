import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const restrictValue = (value, max) => Math.min(Math.max(0, value), max);

const getColor = ({ theme, appearance }) => {
  switch (appearance) {
    case 'danger':
      return theme.danger;
    case 'success':
      return theme.success;
    case 'warning':
      return theme.warning;
    case 'info':
      return theme.info;
    case 'primary':
    default:
      return theme.primaryActive;
  }
};

const ProgressBar = styled(({ max, value, appearance, hideBorder, ...otherProps }) => (
  <div {...otherProps} />
))`
  box-sizing: border-box;
  display: inline-block;
  position: relative;
  overflow: hidden;
  min-width: 240px;
  height: 0.25rem;
  border-radius: 0.125rem;
  background-color: ${p => p.theme.white};
  border: ${p => (p.hideBorder ? 'none' : `solid 1px ${getColor(p)}`)};

  &:before {
    background-color: ${getColor};
    content: '';
    display: block;
    position: absolute;
    top: 0;
    left: 0;
    bottom: 0;
    will-change: width;
    width: ${p => Math.round((100 * restrictValue(p.value, p.max)) / p.max)}%;
    transition: width 1s ease;
  }
`;

ProgressBar.propTypes = {
  max: PropTypes.number.isRequired,
  value: PropTypes.number.isRequired,
  appearance: PropTypes.oneOf(['primary', 'danger', 'success', 'warning', 'info']),
  hideBorder: PropTypes.bool,
};

ProgressBar.defaultProps = {
  hideBorder: false,
  appearance: 'primary',
};

/**
 * @component
 */
export default ProgressBar;
