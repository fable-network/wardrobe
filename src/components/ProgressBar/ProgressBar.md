Value is 0:

```jsx
<ProgressBar max={100} value={0} />
```

Value is 42:

```jsx
<ProgressBar max={100} value={42} />
```

Value is 100:

```jsx
<ProgressBar max={100} value={100} />
```

Different appearance:

```jsx
<ProgressBar max={100} value={42} appearance="warning" />
```

Hide the border:

```jsx
<div style={{ 'backgroundColor': 'lightgrey', padding: '8px' }}>
  <ProgressBar max={100} value={42} hideBorder={true} />
</div>
```
