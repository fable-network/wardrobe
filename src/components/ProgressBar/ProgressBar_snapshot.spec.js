/* eslint-disable */
// This file was automatically generated
import React from 'react';
import 'jest-styled-components';
import ProgressBar from './ProgressBar';


jest.mock('../../charts/HighChart', () => () => <high-charts />);

describe('ProgressBar Component', () => {
  describe('snapshots', () => {
    it('should match snapshot', () => {
    
      const wrapper1 = mount(
        <ProgressBar max={100} value={0} />
      );
      expect(wrapper1).toMatchSnapshot();

      const wrapper2 = mount(
        <ProgressBar max={100} value={42} />
      );
      expect(wrapper2).toMatchSnapshot();

      const wrapper3 = mount(
        <ProgressBar max={100} value={100} />
      );
      expect(wrapper3).toMatchSnapshot();

      const wrapper4 = mount(
        <ProgressBar max={100} value={42} appearance="warning" />
      );
      expect(wrapper4).toMatchSnapshot();

    });
  });
});
