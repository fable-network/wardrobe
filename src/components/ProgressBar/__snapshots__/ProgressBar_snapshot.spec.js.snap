// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar Component snapshots should match snapshot 1`] = `
.c0 {
  box-sizing: border-box;
  display: inline-block;
  position: relative;
  overflow: hidden;
  min-width: 240px;
  height: 0.25rem;
  border-radius: 0.125rem;
  background-color: #ffffff;
  border: solid 1px #034a76;
}

.c0:before {
  background-color: #034a76;
  content: '';
  display: block;
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  will-change: width;
  width: 0%;
  -webkit-transition: width 1s ease;
  transition: width 1s ease;
}

<ProgressBar
  appearance="primary"
  hideBorder={false}
  max={100}
  value={0}
>
  <div
    className="c0"
  />
</ProgressBar>
`;

exports[`ProgressBar Component snapshots should match snapshot 2`] = `
.c0 {
  box-sizing: border-box;
  display: inline-block;
  position: relative;
  overflow: hidden;
  min-width: 240px;
  height: 0.25rem;
  border-radius: 0.125rem;
  background-color: #ffffff;
  border: solid 1px #034a76;
}

.c0:before {
  background-color: #034a76;
  content: '';
  display: block;
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  will-change: width;
  width: 42%;
  -webkit-transition: width 1s ease;
  transition: width 1s ease;
}

<ProgressBar
  appearance="primary"
  hideBorder={false}
  max={100}
  value={42}
>
  <div
    className="c0"
  />
</ProgressBar>
`;

exports[`ProgressBar Component snapshots should match snapshot 3`] = `
.c0 {
  box-sizing: border-box;
  display: inline-block;
  position: relative;
  overflow: hidden;
  min-width: 240px;
  height: 0.25rem;
  border-radius: 0.125rem;
  background-color: #ffffff;
  border: solid 1px #034a76;
}

.c0:before {
  background-color: #034a76;
  content: '';
  display: block;
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  will-change: width;
  width: 100%;
  -webkit-transition: width 1s ease;
  transition: width 1s ease;
}

<ProgressBar
  appearance="primary"
  hideBorder={false}
  max={100}
  value={100}
>
  <div
    className="c0"
  />
</ProgressBar>
`;

exports[`ProgressBar Component snapshots should match snapshot 4`] = `
.c0 {
  box-sizing: border-box;
  display: inline-block;
  position: relative;
  overflow: hidden;
  min-width: 240px;
  height: 0.25rem;
  border-radius: 0.125rem;
  background-color: #ffffff;
  border: solid 1px #f5a623;
}

.c0:before {
  background-color: #f5a623;
  content: '';
  display: block;
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  will-change: width;
  width: 42%;
  -webkit-transition: width 1s ease;
  transition: width 1s ease;
}

<ProgressBar
  appearance="warning"
  hideBorder={false}
  max={100}
  value={42}
>
  <div
    className="c0"
  />
</ProgressBar>
`;
